src/components/signup/EmailConfirmation.js

```diff
@@ -1,104 +1,196 @@
+import React from 'react'
 import normalize from '../../lib/utils/normalizeText'
-import API from '../../lib/API/api'
 import logger from '../../lib/logger/pino-logger'
+import API from '../../lib/API/api'
 import { withStyles } from '../../lib/styles'
+import Icon from '../common/view/Icon'
+import LoadingIndicator from '../common/view/LoadingIndicator'
 import Section from '../common/layout/Section'
-import Wrapper from '../common/layout/Wrapper'
+import { ErrorText } from '../common/form/InputText'
+import OtpInput from '../common/form/OtpInput'
 import CustomWrapper from './signUpWrapper'
-
-type Props = {
-  screenProps: any,
-  navigation: any,
-}
+import type { SignupState } from './SignupState'
 
 const log = logger.child({ from: 'EmailConfirmation' })
 
-const EmailConfirmation = ({ navigation, screenProps, styles }: Props) => {
-  const [globalProfile, setGlobalProfile] = useState({})
-
-  // const API = useWrappedApi()
-  // const userStorage = useWrappedUserStorage()
-  const setLoading = () => log.warn('implement me')
-  useEffect(() => {
-    const { params } = navigation.state
+const DONE = 'DONE'
+const WAIT = 'WAIT'
+const PENDING = 'PENDING'
 
-    const validateEmail = async () => {
-      setLoading(true)
+type Props = {
+  phone: string,
+  data: SignupState,
+  doneCallback: ({ isPhoneVerified: boolean }) => null,
+  screenProps: any,
+}
 
-      // recover user's profile persisted to userStorage in SignupState after sending the email
-      // done before verifying email to have all the user's information available to display
-      const profile = {} //await userStorage.getProfile()
-      setGlobalProfile(profile)
+export type CodeRecord = {
+  codeValidated: boolean,
+  sentCode?: boolean,
+}
 
-      await API.verifyEmail({ code: params.validation })
+type State = CodeRecord & {
+  errorMessage: string,
+  sendingCode: boolean,
+  renderButton: boolean,
+  loading: boolean,
+  code: string | number,
+}
 
-      screenProps.doneCallback({ ...profile, isEmailConfirmed: true })
+class EmailConfirmation extends React.Component<Props, State> {
+  state = {
+    codeValidated: false,
+    sentCode: false,
+    errorMessage: '',
+    sendingCode: false,
+    renderButton: false,
+    resentCode: false,
+    loading: false,
+    code: undefined,
+  }
+  
+  numInputs: number = 6
+  
+  componentDidMount() {}
+  
+  componentDidUpdate() {
+    if (!this.state.renderButton) {
+      this.displayDelayedRenderButton()
     }
-
-    if (params && params.validation) {
-      validateEmail().finally(() => setLoading(false))
+  }
+  
+  displayDelayedRenderButton = () => {
+    setTimeout(() => {
+      this.setState({ renderButton: true })
+    }, 10000)
+  }
+  
+  handleChange = async (code: string | number) => {
+    const codeValue = code.toString()
+  
+    if (codeValue.length === this.numInputs) {
+      this.setState({
+        loading: true,
+        code: code,
+      })
+      try {
+        await this.verifyCode(code)
```

Comment:
the homologous  SMS/OTP code sends the code stringified: https://github.com/GoodDollar/GoodDAPP/blob/b6fb38bcbe1ab7adc4531b2a35d1117d9da2e4eb/src/components/signup/SmsForm.js#L70-L77

I'll prefer to stick to the same typing for the code.